### DISCUSSION TITLE
How to get the current line text in TextArea (without knowing the line number)?

### DISCUSSION THREAD

**Discussion by learnbyexample** [Textualize/textual#4936] [Q&A] [answered]
Hello.

I'm trying to implement smart indent when the Enter key is pressed. Currently, I'm able to mimic the current indentation level using this code:

```python
class ExtendedTextArea(TextArea):
    def _on_key(self, event):
        if event.key == 'enter':
            n = self.get_cursor_line_start_location(smart_home=True)[1]
            self.insert(f'\n{" " * n}')
            event.prevent_default()
```

The above code gets the location of the first non-whitespace character and inserts that many spaces on the new line.

I couldn't figure out how to get the current line's content. That'd help me to add another level of indentation if the current line is a control structure.

PS: If this kinda smart indent is already available in TextArea, let me know 😅

**Comment by darrenburns** [comment, marked as answer]
Use `cursor_line_index, cursor_column_index = self.cursor_location` to get the cursor location indices.

You can use `self.document.get_line(line_index)` to get the line content as a string.

**Reply by learnbyexample** [reply]
Did you mean `self.app.cursor_position`? I got `AttributeError` without `app` prefix.

Then I got `list index out of range`.

```python
class ExtendedTextArea(TextArea):
    def _on_key(self, event):
        if event.key == 'enter':
            cursor_line_index, cursor_column_index = self.app.cursor_position
            line = self.document.get_line(cursor_line_index)
            self.insert(f'\ntest: {line}')
            event.prevent_default()
```

**Reply by darrenburns** [reply]
Sorry, I meant `cursor_location` (edited the original response now).

**Reply by learnbyexample** [reply]
Thanks, that worked!
